File: hphp/hack/src/hhbc/hhas_record_def.ml
(*
 * Copyright (c) 2019, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the "hack" directory of this source tree.
 *
 *)

type t = {
  record_name: Hhbc_id.Record.t;
  record_is_abstract: bool;
  record_base: Hhbc_id.Record.t option;
  record_fields: (string * Hhas_type_info.t * Typed_value.t option) list;
}

let make record_name record_is_abstract record_base record_fields =
  { record_name; record_is_abstract; record_base; record_fields }

let name hhas_record_decl = hhas_record_decl.record_name

let is_abstract hhas_record_decl = hhas_record_decl.record_is_abstract

let base hhas_record_decl = hhas_record_decl.record_base

let fields hhas_record_decl = hhas_record_decl.record_fields

File: hphp/hack/src/hhbc/hhas_record_def.rs
// Copyright (c) Facebook, Inc. and its affiliates.
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the "hack" directory of this source tree.
use hhas_type::Info;
use hhbc_id_rust::record;
use runtime::TypedValue;

#[derive(Debug)]
pub struct Field<'a>(pub &'a str, pub Info, pub Option<TypedValue>);

#[derive(Debug)]
pub struct HhasRecord<'a> {
    pub name: record::Type<'a>,
    pub is_abstract: bool,
    pub base: Option<record::Type<'a>>,
    pub fields: Vec<Field<'a>>,
}
